src/components/multi_select_state.js

```diff
@@ -109,41 +152,76 @@ const withMultiSelectState = WrappedComponent =>
             const index = items.findIndex(item => item.id === id);
             this.setState({ firstItemShiftSelected: index });
           }
+
           const newSelectedItems = this.getNewSelectedItems(id);
-          this.setState({ selectedItems: newSelectedItems }, this.handleChange);
+
+          const newFilteredSelectedItem = newSelectedItems.filter(
+            el => el.id == id
+          );
+
+          this.setState(
+            {
+              selectedItems: newSelectedItems,
+              filteredSelectedItems: [
```

Comment:
this is wrong as it always adds  newly added item to filtered Selected items even when it's not matching filter.
use searchSelectedItemsValue and filterFunction  from properties to calculate it proper and to store in state.
```
const newFilteredSelectedItem = newSelectedItems.filter(filterFunction(searchSelectedItemsValue));
this.setState({
              selectedItems: newSelectedItems,
              filteredSelectedItems: newFilteredSelectedItem
});
```